### DISCUSSION TITLE
code hints with vetur not working

### DISCUSSION THREAD

**Discussion by patialo** [coder/code-server#4905] [Q&A] [answered]
### How can I set my code server to be the same as the code hints of vscode

I use vetur 0.35.0 in these two editors, but the code-server does not have the  vue3 hints  like vscode

I'm sure the hints from vetur, and  have the same setting both (synchronization via SettingSync).

in codeserver 4.0.2
![44d3f63e18289fae4a83d81ea76920e](https://user-images.githubusercontent.com/31113826/155548618-5b838762-e39f-4552-9ae1-89ffb9e54218.png)

in vscode 1.64.2
![4e08cbee277ac149dd53752670bf7e1](https://user-images.githubusercontent.com/31113826/155548743-69a2ab5a-6772-4848-a452-61fd0ccae5d9.png)

**Comment by jsjoeio** [comment, marked as answer]
What extensions are you using and what does your `settings.json` look like?

**Reply by patialo** [reply]
**setting.json**
```
{
	"[jsonc]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[html]": {
		"editor.defaultFormatter": "vscode.html-language-features"
	},
	"[vue]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[javascript]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"todo-tree.tree.showScanModeButton": false,
	"todo-tree.filtering.excludeGlobs": ["**/node_modules", "*.xml", "*.XML"],
	"todo-tree.filtering.ignoreGitSubmodules": true,
	"todohighlight.keywords": [],
	"todo-tree.tree.showCountsInTree": true,
	"todohighlight.keywordsPattern": "TODO:|FIXME:|NOTE:|\\(([^)]+)\\)",
	"todohighlight.defaultStyle": {},
	"todohighlight.isEnable": false,
	"todo-tree.highlights.customHighlight": {
		"BUG": {
			"icon": "bug",
			"foreground": "#F56C6C",
			"type": "line"
		},
		"FIXME": {
			"icon": "flame",
			"foreground": "#FF9800",
			"type": "line"
		},
		"TODO": {
			"foreground": "#FFEB38",
			"type": "line"
		},
		"NOTE": {
			"icon": "note",
			"foreground": "#67C23A",
			"type": "line"
		},
		"RESO": {
			"icon": "info",
			"foreground": "#57ff9d",
			"type": "line"
		},
		"IMP": {
			"icon": "tag",
			"foreground": "#1388fd",
			"type": "line"
		},
		"HACK": {
			"icon": "versions",
			"foreground": "#E040FB",
			"type": "line"
		},
		"XXX": {
			"icon": "unverified",
			"foreground": "#E91E63",
			"type": "line"
		}
	},
	"todo-tree.general.tags": [
		"BUG",
		"HACK",
		"FIXME",
		"TODO",
		"RESO",
		"NOTE",
		"IMP",
		"XXX"
	],
	"todo-tree.general.statusBar": "total",
	"workbench.colorTheme": "One Dark Pro Darker",
	"workbench.startupEditor": "none",
	"workbench.iconTheme": "material-icon-theme",
	"liveServer.settings.port": 5500,
	"liveServer.settings.donotShowInfoMsg": true,
	"typescript.suggest.paths": false,
	"javascript.suggest.paths": false,
	"editor.unicodeHighlight.allowedCharacters": "！：",
	"editor.linkedEditing": true,
	"cSpell.userWords": ["RESO", "vite"],
	"javascript.updateImportsOnFileMove.enabled": "always",
	"explorer.compactFolders": false,
	"files.associations": {
		"*.cjson": "jsonc",
		"*.wxss": "css",
		"*.wxs": "javascript"
	},
	"emmet.includeLanguages": {
		"wxml": "html"
	},
	"minapp-vscode.disableAutoConfig": true,
	"sync.gist": "8c3f254fe3e6135d074fde75919875e6",
	"extensions.autoUpdate": "onlyEnabledExtensions",
	"vetur.completion.autoImport": false,
	"prettier.useTabs": true,
	"prettier.vueIndentScriptAndStyle": true,
	"bracket-pair-colorizer-2.depreciation-notice": false,
	"files.autoSave": "onFocusChange",
	"workbench.editor.splitInGroupLayout": "vertical"
}
```

**extensions**
![f35ca21b66aa2d2e3ee1e6903cf706b](https://user-images.githubusercontent.com/31113826/155660459-ed79271b-b918-4653-9319-c1853b85ab1f.png)
![a5b8bb1689899e97fb27230c278c5ae](https://user-images.githubusercontent.com/31113826/155660464-3d1536e6-874a-48d3-ba97-3da6abf6b660.png)
![fa8527a4f9d5d4924a637980180bedf](https://user-images.githubusercontent.com/31113826/155660466-37944bde-cf7f-40de-bea4-8108f7a96359.png)

**Reply by patialo** [reply]
my code-server run on ubuntu20.04, vscode on windows10

**Reply by jsjoeio** [reply]
To debug this, I would try disabling all extensions and each line in your `settings.json` one-by-one and see if you can figure out what is causing the issue

**Reply by patialo** [reply]
Thanks for your help.
I figured out what's causing the problem，
I've been using the one level up directory of this project and it doesn't read package.json.
This is a mistake that shouldn't be made

**Reply by jsjoeio** [reply]
Ahhh! Hooray! I'm happy you figured it out :) Thanks for following up and letting us know what was wrong too.
